includes/class-ajax-login.php

```diff
@@ -0,0 +1,374 @@
+<?php
+
+/**
+ * Ajax Login and Forgot password handler class
+ *
+ * @since 2.8
+ * @author Tareq Hasan <tareq@wedevs.com>
+ */
+
+class WPUF_Ajax_Login extends WP_Widget {
+
+    function __construct() {
+
+        parent::__construct(
+            'WPUF_Ajax_Login', 
+            __('WPUF Ajax Login', 'wpuf'), 
+            array( 'description' => __( 'Ajax Login widget for WP User Frontend', 'wpuf' ), ) 
+        );
+
+        add_action( 'widgets_init', array( $this, 'wpuf_ajax_login_widget' ) );
+        add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+        add_action( 'wp_ajax_wpuf_ajax_login', array( $this, 'wpuf_ajax_login' ) );
+        add_action( 'wp_ajax_nopriv_wpuf_ajax_login', array( $this, 'wpuf_ajax_login' ) );
+        add_action( 'wp_ajax_wpuf_lost_password', array( $this, 'wpuf_ajax_reset_pass' ) );
+        add_action( 'wp_ajax_nopriv_wpuf_lost_password', array( $this, 'wpuf_ajax_reset_pass' ) );
+        add_action( 'wp_ajax_wpuf_ajax_logout', array( $this, 'wpuf_ajax_logout' ) );
+        add_filter( 'lostpassword_url', array( $this, 'wpuf_lostpassword_url' ));
+    }
+
+    public function enqueue_scripts() {
+        wp_enqueue_script( 'wp_ajax_login', WPUF_ASSET_URI . '/js/wpuf-ajax-login.js', array( 'jquery' ), false, true );
+        
+        wp_localize_script( 'wp_ajax_login', 'wpuf_ajax', array( 
+            'ajaxurl' => admin_url( 'admin-ajax.php' ),
+        ));
+    }
+
+    /**
+     * Ajax Login function
+     *
+     * @return void
+     */
+    public function wpuf_ajax_login() {
+
+        $user_login     = $_POST['log'];  
+        $user_pass      = $_POST['pwd'];
+        $rememberme     = $_POST['rememberme'];
+
+        if ( empty( $user_login ) || empty( $user_pass ) ) {
+            echo json_encode( array('error' => true, 'message'=> __( 'Please fill all form fields', 'wpuf') ) );

```

Comment:
Instead of using `json_encode`, there is a WP native function `wp_send_json_error()` and `wp_send_json_success()`